crates/ruff_server/src/session/settings.rs

```diff
@@ -38,6 +38,22 @@ pub(crate) struct ResolvedEditorSettings {
     pub(super) ignore: Option<Vec<RuleSelector>>,
     pub(super) exclude: Option<Vec<String>>,
     pub(super) line_length: Option<LineLength>,
+    pub(super) configuration_resolution_strategy: ConfigResolutionStrategy,
+}
+
+/// Determines how multiple conflicting configurations should be resolved - in this
+/// case, the configuration from the client settings and configuration from local
+/// `.toml` files (aka 'workspace' configuration).
+#[derive(Clone, Copy, Debug, Deserialize)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[serde(rename_all = "camelCase")]
+pub(crate) enum ConfigResolutionStrategy {
```

Comment:
Nit: `Config` -> `Configuration` for consistency (you use configuration in most other places)